// mui-scroll-trigger/lib/ScrollTrigger.js
"use strict";

exports.__esModule = true;
exports["default"] = ScrollTrigger;

var _react = require("react");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _useScrollTrigger = _interopRequireDefault(require("@mui/material/useScrollTrigger"));

var _Slide = _interopRequireDefault(require("@mui/material/Slide"));

var _Grow = _interopRequireDefault(require("@mui/material/Grow"));

var _Zoom = _interopRequireDefault(require("@mui/material/Zoom"));

var _Collapse = _interopRequireDefault(require("@mui/material/Collapse"));

var _Fade = _interopRequireDefault(require("@mui/material/Fade"));

var _excluded = ["appear", "children", "disabled", "ignoreDirection", "onScroll", "target", "threshold", "transition"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function ScrollTrigger(props) {
  var appear = props.appear,
      children = props.children,
      disabled = props.disabled,
      disableHysteresis = props.ignoreDirection,
      onScroll = props.onScroll,
      target = props.target,
      threshold = props.threshold,
      transition = props.transition,
      rest = _objectWithoutPropertiesLoose(props, _excluded);

  var trigger = (0, _useScrollTrigger["default"])({
    threshold: threshold,
    disableHysteresis: disableHysteresis,
    target: target
  });
  var triggerStatus = onScroll === 'show' ? trigger : !trigger;

  function element() {
    if (transition === 'slide') return _Slide["default"];
    if (transition === 'grow') return _Grow["default"];
    if (transition === 'zoom') return _Zoom["default"];
    if (transition === 'collapse') return _Collapse["default"];
    if (transition === 'fade') return _Fade["default"];
  }

  return (0, _react.createElement)(element(), _extends({
    appear: appear,
    "in": disabled || triggerStatus
  }, rest), children);
}

ScrollTrigger.propTypes = {
  appear: _propTypes["default"].bool,
  children: _propTypes["default"].element.isRequired,
  disabled: _propTypes["default"].bool,
  ignoreDirection: _propTypes["default"].bool,
  onScroll: _propTypes["default"].oneOf(['show', 'hide']),
  target: _propTypes["default"].oneOfType([_propTypes["default"].object, _propTypes["default"].node]),
  threshold: _propTypes["default"].number,
  transition: _propTypes["default"].oneOf(['slide', 'grow', 'zoom', 'collapse', 'fade'])
};
ScrollTrigger.defaultProps = {
  appear: false,
  disabled: false,
  ignoreDirection: false,
  onScroll: 'hide',
  target: window,
  threshold: 100,
  transition: 'grow'
};
module.exports = exports.default;
